src/Output/UnifiedDiffOutputBuilder.php

```diff
@@ -42,84 +62,181 @@ public function getDiff(array $diff): string
             }
         }
 
-        $this->writeDiffChunked($buffer, $diff, $this->getCommonChunks($diff));
+        if (0 !== \count($diff)) {
+            $this->writeDiffHunks($buffer, $diff);
+        }
 
         $diff = \stream_get_contents($buffer, -1, 0);
 
         \fclose($buffer);
 
-        return $diff;
+        // If the last char is not a linebreak: add it.
+        // This might happen when both the `from` and `to` do not have a trailing linebreak
+        $last = \substr($diff, -1);
+
+        return "\n" !== $last && "\r" !== $last
+            ? $diff . "\n"
+            : $diff
+            ;
     }
 
-    // `old` is an array with key => value pairs . Each pair represents a start and end index of `diff`
-    // of a list of elements all containing `same` (0) entries.
-    private function writeDiffChunked($output, array $diff, array $old)
+    private function writeDiffHunks($output, array $diff)
```

Comment:
DRY is heavily violated between 2 `writeDiffHunks` methods

Reply:
`writeDiffHunks` and `writeHunk` are similar in both builders, but not the same.
This is by design; I used the same structure for both builders and isolated the differences in logic needed in the two (private) methods.

As such maintaining the code should be more easy, not more complex. There simply isn't more common logic to subtract from both builders into parent classes or traits or something else.

Reply:
yet, if I compare `writeDiffHunks` of two classes line by line, a lot of lines are the same.

Reply:
Yes, but I don't see a better composition, the difference in output-logic has to go somewhere.

Far more important is to define what changes should be made to the `UnifiedDiffOutputBuilder` and which not. When those specs are clear we can see if there is a better flow to be made.